### PR TITLE
Start using new eval history table

### PR DESCRIPTION
**Pull request by dmjb**
This PR wires the eval history service into the evaluation engine. It is gated behind a feature flag while we experiment with it. A follow on PR will add logging for remediation and alerting.

Relates to #3556

# Summary

***Provide a brief overview of the changes and the issue being addressed. 
Explain the rationale and any background necessary for understanding the changes. 
List dependencies required by this change, if any.***

Fixes #(related issue)

## Change Type

***Mark the type of change your PR introduces:***

- [ ] Bug fix (resolves an issue without affecting existing features)
- [x] Feature (adds new functionality without breaking changes)
- [ ] Breaking change (may impact existing functionalities or require documentation updates)
- [ ] Documentation (updates or additions to documentation)
- [ ] Refactoring or test improvements (no bug fixes or new functionality)

# Testing

***Outline how the changes were tested, including steps to reproduce and any relevant configurations. 
Attach screenshots if helpful.***

# Review Checklist:

- [x] Reviewed my own code for quality and clarity.
- [x] Added comments to complex or tricky code sections.
- [ ] Updated any affected documentation.
- [x] Included tests that validate the fix or feature.
- [ ] Checked that related changes are merged.

### CODE REVIEW

**Review by evankanderson — APPROVED** [review]

File: internal/engine/metrics.go
Comment (evankanderson) [inline internal/engine/metrics.go:1]: This cleanup is not directly related to the rest of the PR, right?  (It's okay, just trying to make sure I understand.)

File: internal/engine/metrics.go [hunk 1]
```diff
@@ -85,27 +85,27 @@ func (e *ExecutorMetrics) CountEvalStatus(
...
 func (e *ExecutorMetrics) CountRemediationStatus(
 	ctx context.Context,
-	status string,
+	status db.RemediationStatusTypes,
 	evalID uuid.UUID,
 	projectID uuid.UUID,
 ) {
 	e.evalCounter.Add(ctx, 1, metric.WithAttributes(
 		attribute.String("profile_id", evalID.String()),
 		attribute.String("project_id", projectID.String()),
```
Comment (evankanderson) [inline internal/engine/metrics.go:94]: You probably _don't_ want to expose these attributes as metric tags, because they will blow up your [metric cardinality](https://opentelemetry.io/docs/specs/otel/metrics/sdk/#cardinality-limits).

Generally, since metrics are aggregated in-memory, you don't want to put non-enum attributes into a metric.

**Review by evankanderson — DISMISSED** [review]
A bit of semi-unrelated commentary on the metrics that I happened to notice.

File: internal/engine/metrics.go
Comment (dmjb) [reply internal/engine/metrics.go:1]: I realized when I wired these metrics into the engine that I was using the wrong type for status.

File: internal/engine/eval_status.go [hunk 2]
```diff
@@ -148,9 +161,9 @@ func (e *Executor) createOrUpdateEvalStatus(
 		return err
 	}
 	status := evalerrors.ErrorAsEvalStatus(params.GetEvalErr())
-	e.metrics.CountEvalStatus(ctx, status, params.ProfileID, params.ProjectID, entityID, entityType)
+	e.metrics.CountEvalStatus(ctx, status, entityType)
 	_, err = e.querier.UpsertRuleDetailsEval(ctx, db.UpsertRuleDetailsEvalParams{
-		RuleEvalID: id,
+		RuleEvalID: evalID,
```
Comment (rdimitrov) [inline internal/engine/eval_status.go:166]: I suppose it's intended but why did we switched from one to another?

**Review by rdimitrov — APPROVED** [review]
